scripts/main/settings.js

```diff
@@ -483,3 +483,47 @@ settings.save_enter = function (e) {
 		},
 	});
 };
+
+settings.viewToken = function () {
+	api.post("User::getCurrent", {}, function (data) {
+		let bodyHtml = "";
+		let enableReset = "";
+		if (data.token === "") {
+			bodyHtml = `<div class='directLinks'><p>${lychee.locale["DISABLED"]}</p></div>`;
+			enableReset = lychee.locale["ENABLE"];
+		} else {
+			bodyHtml = lychee.html`<div class='directLinks'><p><span id="apiToken">${
+				data.token
+			}</span> <a id="button_copy_token" class='basicModal__button' title='${lychee.locale["URL_COPY_TO_CLIPBOARD"]}'>${build.iconic(
+				"copy",
+				"ionicons"
+			)}</a> <a id="button_disable_token" class='basicModal__button' title='${lychee.locale["DISABLE"]}'>${build.iconic("ban")}</a></p></div>`;
+			enableReset = lychee.locale["RESET"];
+		}
+		basicModal.show({
+			body: bodyHtml,
+			buttons: {
+				action: {
+					title: enableReset,
+					fn: function () {
+						api.post("User::resetToken", {});
+						basicModal.close();
+					},
```

Comment:
I don't understand what is supposed to happen here. According to my understanding `resetToken` regenerates the token and returns a new token, correct?

But I don't see that you do something with the result which is returned by `resetToken`. If you essentially throw it away, why not `unsetToken`?

Reply:
This sounds wry. I would have expected to immediately see the result of a call to `resetToken`. I assume I have to check this PR our and test it myself how it works.

But, if we are never going to use the result of `resetToken`, because we need to query for the authenticated user upon the next time the dialog is visited anyway, why does `reset Token` return something at all?